test/Microsoft.Extensions.Internal.Test/StackTraceHelperTest.cs

```diff
@@ -0,0 +1,46 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the Apache License, Version 2.0. See License.txt in the project root for license information.
+
+using System;
+using Microsoft.Extensions.StackTrace.Sources;
+using ThrowingLibrary;
+using Xunit;
+
+namespace Microsoft.Extensions.Internal
+{
+    public class StackTraceHelperTest
+    {
+        [Fact]
+        public void StackTraceHelper_IncludesLineNumbersForFiles()
+        {
+            // Arrange
+            Exception exception = null;
+            try
+            {
+                // Throwing an exception in the current assembly always seems to populate the full stack
```

Comment:
Given the discussion elsewhere in this PR, it would be good to put this same comment in the `Thrower` class itself.